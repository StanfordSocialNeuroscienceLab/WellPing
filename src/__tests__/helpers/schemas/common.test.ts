import {
  QuestionIdSchema,
  QuestionIdSchemaNullable,
} from "../../../helpers/schemas/common";

describe("QuestionIdSchema", () => {
  test("doesn't allow null", () => {
    expect(() => {
      QuestionIdSchema.parse(null);
    }).toThrowErrorMatchingSnapshot();
  });

  test("doesn't allow undefined", () => {
    expect(() => {
      QuestionIdSchema.parse(undefined);
    }).toThrowErrorMatchingSnapshot();
  });

  test("doesn't allow empty", () => {
    expect(() => {
      QuestionIdSchema.parse("");
    }).toThrowErrorMatchingSnapshot();
  });

  test("doesn't allow special characters", () => {
    expect(() => {
      QuestionIdSchema.parse("HELLO WORLD");
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      QuestionIdSchema.parse("hello-world");
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      QuestionIdSchema.parse("测试");
    }).toThrowErrorMatchingSnapshot();
  });

  test("accept", () => {
    expect(() => {
      QuestionIdSchema.parse("1");
    }).not.toThrowError();

    expect(() => {
      QuestionIdSchema.parse("hello");
    }).not.toThrowError();

    expect(() => {
      QuestionIdSchema.parse("hello_world");
    }).not.toThrowError();

    expect(() => {
      QuestionIdSchema.parse("HELLO_WORLD");
    }).not.toThrowError();
  });
});

describe("QuestionIdSchemaNullable", () => {
  test("allow null", () => {
    expect(() => {
      QuestionIdSchemaNullable([]).parse(null);
    }).not.toThrowError();
  });

  test("doesn't allow undefined", () => {
    expect(() => {
      QuestionIdSchemaNullable([]).parse(undefined);
    }).toThrowErrorMatchingSnapshot();
  });

  test("doesn't allow empty", () => {
    expect(() => {
      QuestionIdSchemaNullable([]).parse("");
    }).toThrowErrorMatchingSnapshot();
  });

  test("doesn't allow empty - path info", () => {
    expect(() => {
      QuestionIdSchemaNullable(["myPath"]).parse(undefined);
    }).toThrowErrorMatchingSnapshot();
  });

  test("still allow to be optional", () => {
    expect(() => {
      QuestionIdSchemaNullable([]).optional().parse(undefined);
    }).not.toThrowError();

    expect(() => {
      QuestionIdSchemaNullable(["myPath"]).optional().parse(undefined);
    }).not.toThrowError();

    expect(() => {
      QuestionIdSchemaNullable([]).optional().parse(null);
    }).not.toThrowError();

    expect(() => {
      QuestionIdSchemaNullable([]).optional().parse("hello");
    }).not.toThrowError();
  });
});
