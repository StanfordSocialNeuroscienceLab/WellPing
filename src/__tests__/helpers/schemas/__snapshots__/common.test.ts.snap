// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionIdSchema doesn't allow empty 1`] = `
"2 validation issue(s)

  Issue #0: too_small at 
  Should be at least 1 characters

  Issue #1: custom_error at 
  Question ID can only include letters, numbers, \\"_\\", \\"[\\", and \\"]\\".
"
`;

exports[`QuestionIdSchema doesn't allow null 1`] = `
"1 validation issue(s)

  Issue #0: invalid_type at 
  Expected string, received null
"
`;

exports[`QuestionIdSchema doesn't allow special characters except _, [, and ] 1`] = `
"1 validation issue(s)

  Issue #0: custom_error at 
  Question ID can only include letters, numbers, \\"_\\", \\"[\\", and \\"]\\".
"
`;

exports[`QuestionIdSchema doesn't allow special characters except _, [, and ] 2`] = `
"1 validation issue(s)

  Issue #0: custom_error at 
  Question ID can only include letters, numbers, \\"_\\", \\"[\\", and \\"]\\".
"
`;

exports[`QuestionIdSchema doesn't allow special characters except _, [, and ] 3`] = `
"1 validation issue(s)

  Issue #0: custom_error at 
  Question ID can only include letters, numbers, \\"_\\", \\"[\\", and \\"]\\".
"
`;

exports[`QuestionIdSchema doesn't allow special characters except _, [, and ] 4`] = `
"1 validation issue(s)

  Issue #0: custom_error at 
  Question ID can only include letters, numbers, \\"_\\", \\"[\\", and \\"]\\".
"
`;

exports[`QuestionIdSchema doesn't allow undefined 1`] = `
"1 validation issue(s)

  Issue #0: invalid_type at 
  Required
"
`;
